### PR TITLE
Feat: Add Logger

### PR DESCRIPTION
**Pull request by AmanRaj1608**
Instead of adding a new package for adding logger in all packages, let's reuse the Logger class.

### CODE REVIEW

File: packages/account/src/BiconomySmartAccountV2.ts [hunk 1]
```diff
@@ -127,19 +128,20 @@ export class BiconomySmartAccountV2 extends BaseSmartContractAccount {
...
+    const config = {
+      ...biconomySmartAccountConfig,
+    };
 
     // Note: If no module is provided, we will use ECDSA_OWNERSHIP as default
-    if (biconomySmartAccountConfig.defaultValidationModule) {
-      instance.defaultValidationModule = biconomySmartAccountConfig.defaultValidationModule;
-    } else {
-      instance.defaultValidationModule = await ECDSAOwnershipValidationModule.create({
+    if (!biconomySmartAccountConfig?.defaultValidationModule) {
```
Comment (joepegler) [inline packages/account/src/BiconomySmartAccountV2.ts:136]: My logic was slightly off here.

```
if (!config?.defaultValidationModule) {
  const newModule = await ECDSAOwnershipValidationModule.create({
    signer: biconomySmartAccountConfig.signer!,
  });
  config.defaultValidationModule = newModule;
}
config.activeValidationModule = config?.activeValidationModule ?? config.defaultValidationModule;

return new BiconomySmartAccountV2(config);
```

**Review by joepegler — CHANGES_REQUESTED** [review]

File: packages/account/src/BiconomySmartAccountV2.ts (hunk 1 above)
Comment (AmanRaj1608) [reply packages/account/src/BiconomySmartAccountV2.ts]: Yeah makes sense, updated the code

**Review by joepegler — APPROVED** [review]

**Review by joepegler — CHANGES_REQUESTED** [review]
You'll also need to set it in the constructor the same way you do the others: 

```    this.defaultValidationModule = biconomySmartAccountConfig.defaultValidationModule;
    this.activeValidationModule = biconomySmartAccountConfig.activeValidationModule;

```

**Review by joepegler — APPROVED** [review]

**Comment by VGabriel45** [comment]
Can we have the Logger class just in one place and use it from there instead of adding it in every module where we need it ?

**Comment by AmanRaj1608** [comment]
@VGabriel45 This was there previously in `common` package. Adding this will introduce again another package which we need to deploy. If we have it in one package and import in another it can introduce circular dependecies
account importing paymaster for paymaster class and paymaster importing account for common package.

So solution I think of
- Use an external logger class and add as dependency in all class
- Re-use code but this is much lighter than any package and does the job

**Review by VGabriel45 — APPROVED** [review]

**Pull request merged**